src/collections/_documentation/platforms/javascript/react.md

```diff
@@ -36,40 +45,104 @@ In development mode, React will rethrow errors caught within an error boundary.
   level="warning"
 %}
 
+In the example below, when the `<Example />` component hits an error, the `<Sentry.ErrorBoundary>` component will send data about that error and the component tree to Sentry, and open a user feedback dialog, and render a fallback UI.
+
 ```jsx
-import React, { Component } from 'react';
-import * as Sentry from '@sentry/browser';
-
-class ExampleBoundary extends Component {
-    constructor(props) {
-        super(props);
-        this.state = { eventId: null };
-    }
-
-    static getDerivedStateFromError() {
-      return { hasError: true };
-    }
-
-    componentDidCatch(error, errorInfo) {
-      Sentry.withScope((scope) => {
-          scope.setExtras(errorInfo);
-          const eventId = Sentry.captureException(error);
-          this.setState({eventId});
-      });
-    }
-
-    render() {
-        if (this.state.hasError) {
-            //render fallback UI
-            return (
-              <button onClick={() => Sentry.showReportDialog({ eventId: this.state.eventId })}>Report feedback</button>
-            );
-        }
-
-        //when there's not an error, render children untouched
-        return this.props.children;
-    }
+import React from 'react';
+import * as Sentry from '@sentry/react';
+
+import {Example} from '../example';
+
+function FallbackComponent() {
+  return (
+    <div>An error has occured</div>
+  )
+}
+
+class App extends React.Component {
+  render() {
+    return (
+      <Sentry.ErrorBoundary fallback={FallbackComponent} showDialog>
+        <Example />
+      </Sentry.ErrorBoundary>
+    )
+  }
 }
 
-export default ExampleBoundary
+export default App;
 ```
+
+### ErrorBoundary Options
+
+The ErrorBoundary component exposes a variety props that can be passed in for extra configuration. There are no required options, but it is highly recommended that a fallback component be set.
+
+#### `showDialog` (boolean)
+
+If a [Sentry User Feedback Widget]({%- link _documentation/enriching-error-data/user-feedback.md -%}?platform={{ include.platform }}) should be rendered when the Error Boundary catches an error.
+
+#### `dialogOptions` (Object)
+
+Options that are passed into the Sentry User Feedback Widget. See all possible customization options [here]({%- link _documentation/enriching-error-data/user-feedback.md -%}?platform={{ include.platform }}#customizing-the-widget)
+
+#### `fallback` (React.ReactNode or Function)
+
+A fallback component that gets rendered when the error boundary encounters an error. Can either provide a React Component, or a function that returns React Component as a valid fallback prop. If a function is provided, the function will be called with the error and the component stack at time of error, and a function that resets the error boundary to it's regular state.
```

Comment:
```suggestion
A fallback component that gets rendered when the error boundary encounters an error. It can either provide a React Component, or a function that returns React Component as a valid fallback prop. If you provide a function, Sentry will call the function with the error and the component stack at the time of the error. 
```
The original last sentence was difficult to parse. I broke up the sentence and deleted second half of the sentence because I couldn't understand what it was trying to communicate. Feel free to rephrase what I wrote.

Reply:
Yes it was, good point. I reworded based on your suggestion. Thanks!